packages/profiling-browser/src/integration.ts

```diff
@@ -0,0 +1,129 @@
+import { getCurrentHub } from '@sentry/core';
+import type { Event, EventProcessor, Hub, Integration } from '@sentry/types';
+import { logger } from '@sentry/utils';
+import { LRUMap } from 'lru_map';
+
+import type { ProcessedJSSelfProfile } from './jsSelfProfiling';
+import type { ProfiledEvent } from './utils';
+import { createProfilingEventEnvelope } from './utils';
+
+// We need this integration in order to actually send data to Sentry. We hook into the event processor
+// and inspect each event to see if it is a transaction event and if that transaction event
+// contains a profile on it's metadata. If that is the case, we create a profiling event envelope
+// and delete the profile from the transaction metadata.
+export const PROFILING_EVENT_CACHE = new LRUMap<string, Event>(20);
+/**
+ * Browser profiling integration. Stores any event that has contexts["profile"]["profile_id"]
+ * This exists because we do not want to await async profiler.stop calls as transaction.finish is called
+ * in a synchronous context. Instead, we handle sending the profile async from the promise callback and
+ * rely on being able to pull the event from the cache when we need to construct the envelope. This makes the
+ * integration less reliable as we might be dropping profiles when the cache is full.
+ */
+export class BrowserProfilingIntegration implements Integration {
+  public readonly name: string = 'BrowserProfilingIntegration';
+  public getCurrentHub?: () => Hub = undefined;
```

Comment:
m: Any reason this is public?

Reply:
Not anymore, I can change that. We also dont need the getCurrentHub anymore